pkg/controller/egress/controller.go

```diff
@@ -449,3 +456,61 @@ func (c *EgressController) enqueueEgresses(poolName string) {
 		c.queue.Add(egress.Name)
 	}
 }
+
+func (c *EgressController) updateEgressAllocatedCondition(egress *egressv1beta1.Egress, err error) {
+	var desiredCondition *egressv1beta1.EgressCondition
+	if egress.Spec.ExternalIPPool != "" {
+		if err == nil {
+			desiredCondition = &egressv1beta1.EgressCondition{
+				Type:               egressv1beta1.IPAllocated,
+				Status:             v1.ConditionTrue,
+				Reason:             "Allocated",
+				Message:            "EgressIP is successfully allocated",
+				LastTransitionTime: metav1.Now(),
+			}
+		} else {
+			desiredCondition = &egressv1beta1.EgressCondition{
+				Type:               egressv1beta1.IPAllocated,
+				Status:             v1.ConditionFalse,
+				Reason:             "AllocationError",
+				Message:            fmt.Sprintf("Cannot allocate EgressIP from ExternalIPPool %s due to: %v", egress.Spec.ExternalIPPool, err),
+				LastTransitionTime: metav1.Now(),
+			}
+		}
+	}
+
+	toUpdate := egress.DeepCopy()
+	var updateErr, getErr error
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		actualCondition, index := getIPAllocatedCondition(toUpdate)
+		if actualCondition != nil && actualCondition.Status == desiredCondition.Status && actualCondition.Reason == desiredCondition.Reason && actualCondition.Message == desiredCondition.Message {
```

Comment:
`desiredCondition` could be `nil`, the program would panic here. You can define a function `compareConditionIgnoringTimestamp` to compare `actualCondition` and `desiredCondition`.